### PR TITLE
[AMD] Avoid wide page-table expansion in chunked DSA prefill

### PR DESCRIPTION
**Pull request by pbkowalski**
# [AMD] Avoid wide page-table expansion in chunked DSA prefill

## Motivation

Long-context DSA prefill splits the logits into row chunks when the full FP32
logits tensor would exceed the memory limit. For example, a 32,768 by 60,000
tensor needs about 7.9 GiB.

For each chunk, the PAGED path currently expands the page table with
`page_table_1[batch_idx_list]` before selecting top-k entries. This copies every
page-table column for every row, even though only the selected top-k indices are
needed.

## Modifications

- Select the logical top-k indices before reading the page table.
- Add a Triton row-map kernel that uses `batch_idx_list` to find the page-table
  row for each token.
- Map only the selected logical indices to physical KV-cache slots.
- Read the page-table row stride at runtime so different context lengths do not
  compile separate Triton modules.

At 60,000 tokens with top-k 2,048, this maps about 30 times less page-table data.
The decode path is unchanged.

[SGLang PR #37889](https://github.com/sgl-project/sglang/pull/37889) handles
eligible unchunked rows with the fused top-k v2 kernel. Its chunked fallback
still uses the wide page-table expansion. This PR handles that fallback when
`batch_idx_list` is present.

## Accuracy Tests

The row-map output matches direct PyTorch indexing for:

- 3 requests, 17 rows, page-table width 257, and top-k 64;
- 16 requests, 32 rows, page-table width 60,003, and top-k 2,048.

The tests also check that `batch_idx_list` selects the row-map path.

```bash
python3 -m pytest \
  test/registered/kernel/attention/test_dsa_prefill_rowmap.py -v
```

## Speed Tests and Profiling

Hardware: AMD MI355X, TP4, GLM-5.2-MXFP4, 60,000 input tokens, concurrency 16.

- Time in `aten::index` decreased from 1.584 seconds to 0.004 seconds.
- Total GPU kernel time decreased by 4.4%.
- Output throughput increased by 1.5%.
- Mean time to first token decreased by 2.3%.

## Checklist

- [x] Format and code-style checks pass.
- [x] Tests cover small and long-context page tables.
- [x] The test is registered for AMD and CUDA CI.
- [x] The fused decode top-k path is unchanged.
- [x] No user-facing documentation change is required.

---
### CI States

Latest PR Test (Base): :x: [Run #34595925176](https://github.com/sgl-project/sglang/actions/runs/34595925176)
Latest PR Test (Extra): :x: [Run #34595925028](https://github.com/sgl-project/sglang/actions/runs/34595925028)
Latest PR Test (AMD ROCm 10): :x: [Run #34595925085](https://github.com/sgl-project/sglang/actions/runs/34595925085)

### CODE REVIEW

File: python/sglang/srt/layers/attention/dsa/dsa_topk_backend.py [hunk 1]
```diff
@@ -160,6 +160,22 @@ def topk_transform(
...
+                        transform_index_page_table_row_map,
+                    )
+
+                    logical_topk = self.topk_func(
+                        logits,
+                        lengths,
+                        topk,
+                        row_starts=row_starts,
+                    )
+                    return transform_index_page_table_row_map(
```
Comment (rkarhila-amd) [inline python/sglang/srt/layers/attention/dsa/dsa_topk_backend.py]: aiter dsa_topk_backend now has this new kernel launch.
When does it get triggered? Does is affect small workloads?
